### ISSUE TITLE
[Feature request] Add Wifi direct support with QR

### ISSUE THREAD

**Issue by decipher3114** [localsend/localsend#189]
### Is your feature request related to a problem? Please describe.

Sometimes, when I want to share file from my Phone (on mobile data) to PC (on wifi), I had to connect my phone to wifi.

### Describe the solution you'd like

Add support of quick "Wifi Direct" generation and detection of the device with network discovery. And connection of the devices with Wifi Direct.

### Describe alternatives you've considered

Use of QR code to identify device is also a good way to implement this.

### Additional context

Same as Shareit and Xander used to do.

**Comment by dnkmmr69420** [comment]
I would love to see this implemented

**Comment by 5wHN28Dg** [comment]
I really hope this gets implemented to, WiFi Direct would allow for an even faster speeds by removing a potential bottleneck (your LAN) and make the app even more convenient to use (no need to worry about setting a LAN) and allow for better experience (staying connected to the internet while transferring files or not needing to switch to WiFi as OP mentioned)

**Comment by 0x000005** [comment]
这个非常难实现吗，为什么全网都没有利用wifi-direct的开放软件😭😭😭

**Comment by 0-BlackSpectrum-0** [comment]
Yes, please add this feature. It would be especially useful when I want to transfer something while I'm outside and there are only two devices, and I'd like to transfer locally without using those chat apps.

**Comment by theguywiththeproblem** [comment]
I see that recently Apple has been forced to support wi-fi aware (https://developer.apple.com/documentation/WiFiAware). Would this be able to provide any benefit to p2p sharing (especially as a full alternative to airdrop for iphones) or would that require redoing too much of the existing code?

**Comment by masch-me** [comment]
Hi! First of all I’d like to thank the team of LocalSend as well - awesome app, no ads - thank you for developing such an incredible piece of software, and as Open Source!
I also have the use case to share cross iOS and Android P2P when I’m outside - like BlackSpectrum.
@Tienisto Sorry, if this isn’t the right place, but can I sponsor this feature? If it’s possible, how much would it cost to develop real P2P transfers?

**Comment by Tienisto** [comment]
@masch-me Thanks for your support. WiFi Aware is already planned so you don't need to sponsor this feature explicitly. Unfortunately, I cannot tell the time it needs (and therefore, the costs) because WiFi Aware is quite new and the libraries (especially in Rust) do not exist yet. This year, I will focus on LocalSend though so you can expect more progress this year.

**Comment by masch-me** [comment]
@Tienisto Thank you for the update! That's great news! Regarding the sponsoring: Thank you also for the information. I will donate to the project instead. Again, thank you for the time and effort you put in this project!

**Comment by HardikRa** [comment]
@Tienisto I've played around with WiFi Aware and the user flow and pairing specially when done cross platform feels like bad UX, I have found a way and built a connection layer to do it via QR with less friction, for iOS and Android currently. I developed it and it is working on my company's application pretty well. I'd love to discuss or contribute this to Localsend as a feature. It uses a combination of SoftAP, BLE and AWDL and does so in a very seamless manner. Happy to discuss more details or share a demo in case it makes sense. 
Also, it's a utility at the OS connection layer and so does not require changes to the Rust core
